code/API_definitions/connectivity-insights-subscriptions.yaml

```
@@ -0,0 +1,1265 @@
+
+    ### Notification callback
+
+      This endpoint describes the event notification received on subscription
+      listener side when the event occurred. As for subscription, detailed
+      description of the event notification is provided in the CAMARA API design guideline document.
+
+      **WARNING**: This callback endpoint must be exposed on the listener
+      side as `POST /{$request.body#/sink}`will be posted by the
+      connectivity insights resource server to the webhook url provided
+      by notification listener.
+
+    # Authorization and Authentication
+
+      [Camara Security and Interoperability Profile](https://github.com/
+      camaraproject/IdentityAndConsentManagement/blob/r0.2.0-rc.2/
+      documentation/CAMARA-Security-Interoperability.md) provides details on
+      how a client requests an access token.
+
+      Which specific authorization flows are to be used will be determined
+      during onboarding process, happening between the API Client and the Telco
+      Operator exposing the API, taking into account the declared purpose for
+      accessing the API, while also being subject to the prevailing legal
+      framework dictated by local legislation.
+
+      It is important to remark that in cases where personal user data is
+      processed by the API, and users can exercise their rights through
+      mechanisms such as opt-in and/or opt-out, the use of 3-legged access
+      tokens becomes mandatory. This measure ensures that the API remains in
+      strict compliance with user privacy preferences and regulatory
+      obligations, upholding the principles of transparency and user-centric
+      data control.
+
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+
+externalDocs:
+  description: Product documentation at CAMARA.
+  url: https://github.com/camaraproject/ConnectivityInsights
+
+servers:
+  - url: "{apiRoot}/connectivity-insights-subscriptions/v0.4rc1/check-network-quality"
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: |
+          API root, defined by service provider, e.g.
+          `api.example.com` or `api.example.com/somepath`
+
+tags:
+  - name: Connectivity Insights Subscriptions
+    description: |
+      Create and manage a subscription to receive periodic connectivity
+      insights
+
+paths:
+  /subscriptions:
+    post:
+      tags:
+        - Connectivity Insights Subscriptions
+      description: "Create a Connectivity insights subscription for a device"
+      summary: "Create a Connectivity insights subscription for a device"
+      operationId: createSubscription
+      parameters:
+        - $ref: "#/components/parameters/x-correlator"
+      security:
+        - openId:
+            - connectivity-insights-subscriptions:create
```

https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#apis-which-deal-with-explicit-subscriptions

```suggestion
            - connectivity-insights-subscriptions:org.camaraproject.connectivity-insights-subscriptions.v0.network-quality-met:create
            - connectivity-insights-subscriptions:org.camaraproject.connectivity-insights-subscriptions.v0.network-quality-not-met:create

```